src/etos_lib/kubernetes/etos.py

```diff
@@ -0,0 +1,135 @@
+# Copyright Axis Communications AB.
+#
+# For a full list of individual contributors, please see the commit history.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Kubernetes client for ETOS custom resources."""
+import os
+import logging
+from pathlib import Path
+from typing import Optional
+from pydantic import BaseModel
+from kubernetes import config
+from kubernetes.client import api_client
+from kubernetes.dynamic import DynamicClient
+from kubernetes.dynamic.resource import Resource as DynamicResource, ResourceInstance
+from kubernetes.dynamic.exceptions import NotFoundError
+
+config.load_config()
+NAMESPACE_FILE = Path("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
+
+
+class NoNamespace(Exception):
+    """NoNamespace exception is raised when ETOS could not determine the current namespace."""
+
+
+class Resource:
+    """Resource is the base resource client for ETOS custom resources."""
+
+    client: DynamicResource
+    namespace: str = "default"
+
+    def get(self, name: str) -> Optional[ResourceInstance]:
+        """Get a resource from Kubernetes by name."""
+        try:
+            resource = self.client.get(name=name, namespace=self.namespace)  # type: ignore
+        except NotFoundError:
+            resource = None
+        return resource
+
+    def delete(self, name: str) -> bool:
+        """Delete a resource by name."""
+        if self.client.delete(name=name, namespace=self.namespace):  # type: ignore
```

Comment:
Retries?  Handle more exceptions?

Reply:
No retries. Up to the user of this client to retry.

Reply:
In other cases where we do things in kubernetes we use the python client directly which raises exceptions on errors where we need to handle it.
If we add error handling here then we would have two behaviors when working with Kubernetes.
1. The user has to do its own error handling
2. The user cannot do any error handling

And considering how we've tried to remove error handling from the etos library (see the http lib) due to it being really difficult to deal with all different error handling and retry strategies that each service wants.

I vote for no error handling here, and that each service implements its own.

Reply:
Ok.

Reply:
But please add this explanation to the code as a comment.

Reply:
Done